server/service/src/stocktake_line/validate.rs

```diff
@@ -66,6 +66,7 @@ pub fn check_reason_is_valid(
                 InventoryAdjustmentReasonFilter::new()
                     .r#type(InventoryAdjustmentType::Positive.equal_to())
                     .is_active(true)
+                    .include_system_reasons(true)
```

Comment:
Include the system reasons as valid inventory adjustment reasons here, otherwise defaulting to filtering them out everywhere else